### PR TITLE
Mark classes final, or hide ctor for testing classes

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/OpenTelemetryAssertions.java [hunk 1]
```diff
@@ -26,5 +26,5 @@ public static SpanDataAssert assertThat(SpanData spanData) {
     return new SpanDataAssert(spanData);
   }
 
-  protected OpenTelemetryAssertions() {}
+  private OpenTelemetryAssertions() {}
```
Comment (jkwatson) [inline sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/OpenTelemetryAssertions.java:29]: I feel like there must have been a good reason for this to be `protected` explicitly. Is this being extended somewhere like the instrumentation project.

Comment (bogdandrutu) [reply sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/OpenTelemetryAssertions.java:29]: Not used https://github.com/open-telemetry/opentelemetry-java-instrumentation/search?q=OpenTelemetryAssertions

Comment (jkwatson) [reply sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/OpenTelemetryAssertions.java:29]: If it turns out there is a good reason for this (maybe in one of the custom agent distributions?), we can very easily open it back up. 👍

**Review by jkwatson — APPROVED** [review]

File: sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/OpenTelemetryAssertions.java (hunk 1 above)
Comment (anuraaga) [inline sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/OpenTelemetryAssertions.java:29]: @bogdandrutu @jkwatson Yeah it's normal practice for the entrypoint into assertions to be extendable - it's how a user would be able to add their own custom assertions while using these too.

Comment (jkwatson) [reply sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/OpenTelemetryAssertions.java:29]: I'm totally fine opening these back up. It's testing code, not production code, and I think we should make it as easy as possible for our users to write tests for otel and extend as needed.

**Pull request merged**